src/TwoSum.java

```diff
@@ -6,7 +6,7 @@ public class TwoSum {
     public static int[] twoSum(int[] nums, int target) {
         int n = nums.length;
         HashMap<Integer, Set<Integer>> vals = new HashMap<>();
```

Comment:
I know I said we need to use Set<Integer> as value, but on second thought we don't. 
The function can be simplified to this: 

public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> indexMap = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            var idx = indexMap.get(target - nums[i]);
            if(idx != null) {
                return new int[] { idx, i };
            }
            indexMap.put(nums[i], i);
        }
        return null;
    }

Reply:
indexMap.put results updating index value if value is repeated, so the index is the very last occurrence of value in array. Method putIfAbsent() for HashMap resolves this issue